src/averageAges.js

```diff
@@ -55,10 +63,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredPeople = onlyWithSon ? people.filter(
+    person => person.sex === 'm'
+  )
+    : people;
+
+  const ageDiffs = filteredPeople.map(person => {
+    const mother = people.find(m => m.name === person.mother);
+
+    return mother ? person.born - mother.born : 0;
+  });
+
+  const validAgeDiffs = ageDiffs.filter(diff => diff !== 0);
+
+  const totalAgeDiff = validAgeDiffs.reduce((sum, ageDiff) => sum + ageDiff, 0);
```

Comment:
looks like we have the same logic for finding the average age. Let's move it to helper and reuse